### ISSUE TITLE
benchmark error

### ISSUE THREAD

**Issue by bradleyashton** [JuliaComputing/ArrayFire.jl#131]
If I do 
```julia
julia> include("benchmark.jl")
INFO: Warmup done!
INFO: Matmul
ERROR: LoadError: ArrayFire Error (998) : Internal error
 in throwAFerror(::Int32) at /home/abradley/.julia/v0.5/ArrayFire/src/error.jl:50
 in af_sync(::Int64) at /home/abradley/.julia/v0.5/ArrayFire/src/wrap.jl:1233
 in matmul(::ArrayFire.AFArray{Float32,2}, ::ArrayFire.AFArray{Float32,2}) at /home/abradley/ArrayFire.jl/benchmarks/benchmark.jl:21
 in macro expansion at ./util.jl:226 [inlined]
 in benchmark() at /home/abradley/ArrayFire.jl/benchmarks/benchmark.jl:68
 in include_from_node1(::String) at ./loading.jl:488
while loading /home/abradley/ArrayFire.jl/benchmarks/benchmark.jl, in expression starting on line 107

julia> include("nmf_benchmark.jl")
INFO: Warmup
ERROR: LoadError: ArrayFire Error (998) : Internal error
 in throwAFerror(::Int32) at /home/abradley/.julia/v0.5/ArrayFire/src/error.jl:50
 in af_sync(::Int64) at /home/abradley/.julia/v0.5/ArrayFire/src/wrap.jl:1233
 in nmf_gpu(::ArrayFire.AFArray{Float32,2}, ::Int64, ::Int64) at /home/abradley/ArrayFire.jl/benchmarks/nmf_benchmark.jl:19
 in nmf_gpu(::ArrayFire.AFArray{Float32,2}, ::Int64) at /home/abradley/ArrayFire.jl/benchmarks/nmf_benchmark.jl:13
 in include_from_node1(::String) at ./loading.jl:488
while loading /home/abradley/ArrayFire.jl/benchmarks/nmf_benchmark.jl, in expression starting on line 27
```

```julia
julia> versioninfo()
Julia Version 0.5.0
Commit 3c9d753 (2016-09-19 18:14 UTC)
Platform Info:
  System: Linux (x86_64-linux-gnu)
  CPU: Intel(R) Core(TM) i7-4770 CPU @ 3.40GHz
  WORD_SIZE: 64
  BLAS: libopenblas (NO_LAPACK NO_LAPACKE DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: liblapack.so.3
  LIBM: libopenlibm
  LLVM: libLLVM-3.7.1 (ORCJIT, haswell)
```

**Comment by ranjanan** [comment]
@bradleyashton can you start up julia by doing `AF_PRINT_ERRORS=1 julia` and then running this again? Do you get any different output?

**Comment by bradleyashton** [comment]
```julia
julia> include("benchmark.jl")
ERROR: could not open file /home/abradley/benchmark.jl
 in include_from_node1(::String) at ./loading.jl:488

julia> include("nmf_benchmark.jl")
ERROR: could not open file /home/abradley/nmf_benchmark.jl
 in include_from_node1(::String) at ./loading.jl:488
```

**Comment by ranjanan** [comment]
Umm, this looks like the error you'd get if the files weren't in the same directory. Are you sure that you're the `benchmarks` folder and this is the error you're getting?

**Comment by bradleyashton** [comment]
oops, updated

**Comment by ranjanan** [comment]
Hmmm, interesting. Can you paste the output of `AFInfo()`?

**Comment by bradleyashton** [comment]
```julia
julia> AFInfo()
ArrayFire v3.5.0 (CUDA, 64-bit Linux, build e59cdbc)
Platform: CUDA Toolkit 7, Driver: 367.57
[0] Tesla K40c, 11440 MB, CUDA Compute 3.5
-1- Tesla K40c, 11440 MB, CUDA Compute 3.5
```

**Comment by ranjanan** [comment]
Could you run the following code on the REPL and post the output?
```julia
function run(sz::Int)
    a = rand(AFArray{Float64}, sz, sz)
    b = rand(AFArray{Float64}, sz, sz)
    for i = 1:10
        r = a * b
        ArrayFire.eval(r)
    end
    sync()
end
run(100)
````

**Comment by bradleyashton** [comment]
```julia
julia> run(100)
In function int cuda::DeviceManager::setActiveDevice(int, int)
In file src/backend/cuda/platform.cpp:428
CUDA Error (10): invalid device ordinal


ERROR: ArrayFire Error (998) : Internal error
 in throwAFerror(::Int32) at /home/abradley/.julia/v0.5/ArrayFire/src/error.jl:50
 in af_sync(::Int64) at /home/abradley/.julia/v0.5/ArrayFire/src/wrap.jl:1233
 in run(::Int64) at ./REPL[8]:8
```

**Comment by ranjanan** [comment]
I'm not sure actually. This is an error internal to the c++ library. @pavanky, do you know what's happening here?

**Comment by pavanky** [comment]
@bradleyashton what GPU do you have? The error you are seeing occurs when you are running arrayfire on a GPU it is not compiled for.

**Comment by ranjanan** [comment]
According to https://github.com/JuliaComputing/ArrayFire.jl/issues/131#issuecomment-275068688, he has Tesla K40c.

**Comment by pavanky** [comment]
@ranjanan Thanks

@bradleyashton 

1) Did you compile arrayfire on a different machine before moving it to this machine ?
2) Did you just update a driver and / or the Linux kernel ?

**Comment by bradleyashton** [comment]
ok, if I run each line of the function, I get:
```julia
julia> sync()
In function int cuda::DeviceManager::setActiveDevice(int, int)
In file src/backend/cuda/platform.cpp:428
CUDA Error (10): invalid device ordinal


ERROR: ArrayFire Error (998) : Internal error
 in throwAFerror(::Int32) at /home/abradley/.julia/v0.5/ArrayFire/src/error.jl:50
 in af_sync(::Int64) at /home/abradley/.julia/v0.5/ArrayFire/src/wrap.jl:1233
 in sync() at /home/abradley/.julia/v0.5/ArrayFire/src/backend.jl:89
```

to answer the above questions

`1. Did you compile arrayfire on a different machine before moving it to this machine ?`

No, compiled from source according to 

https://github.com/arrayfire/arrayfire/wiki/Build-Instructions-for-Linux

with `cmake .. -DCMAKE_BUILD_TYPE=Release -DBUILD_CUDA=On`

`2. Did you just update a driver and / or the Linux kernel ?`

No. In light of this I thought I would rerun the `arrayfire` tests in case something has changed, but all pass ok.

**Comment by pavanky** [comment]
@bradleyashton It is likely the Julia wrapper is picking up an incompatible arrayfire library somehow.

Can you launch Julia using `AF_SHOW_LOAD_PATH julia` and run the commands again?

**Comment by bradleyashton** [comment]
```shell
abradley@kolmogorov:~$ AF_SHOW_LOAD_PATH=1 julia
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.5.0 (2016-09-19 18:14 UTC)
 _/ |\__'_|_|_|\__'_|  |  
|__/                   |  x86_64-linux-gnu
```
```julia
julia> using ArrayFire

julia> function run(sz::Int)
           a = rand(AFArray{Float64}, sz, sz)
           b = rand(AFArray{Float64}, sz, sz)
           for i = 1:10
               r = a * b
               ArrayFire.eval(r)
           end
           sync()
       end
run (generic function with 1 method)

julia> run(100)
Using libafcpu.so.3 from system path
Using libafopencl.so.3 from system path
Using libafcuda.so.3 from system path
ERROR: ArrayFire Error (998) : Internal error
 in throwAFerror(::Int32) at /home/abradley/.julia/v0.5/ArrayFire/src/error.jl:50
 in af_sync(::Int64) at /home/abradley/.julia/v0.5/ArrayFire/src/wrap.jl:1233
 in run(::Int64) at ./REPL[2]:8
```

**Comment by pavanky** [comment]
@bradleyashton 

```
sudo updatedb
locate libafcuda.so.3
```

**Comment by bradleyashton** [comment]
```shell
abradley@kolmogorov:~$ sudo updatedb
[sudo] password for abradley: 
abradley@kolmogorov:~$ locate libafcuda.so
/home/abradley/arrayfire/build/src/backend/cuda/libafcuda.so
/home/abradley/arrayfire/build/src/backend/cuda/libafcuda.so.3
/home/abradley/arrayfire/build/src/backend/cuda/libafcuda.so.3.5.0
/usr/local/lib/libafcuda.so
/usr/local/lib/libafcuda.so.3
/usr/local/lib/libafcuda.so.3.5.0
abradley@kolmogorov:~$ 
```

**Comment by pavanky** [comment]
@bradleyashton are you able to run `helloworld_cuda` example ?

**Comment by bradleyashton** [comment]
I ran the file I found here: 
[https://www.pdc.kth.se/resources/computers/historical-computers/zorn/how-to/how-to-compile-and-run-a-simple-cuda-hello-world](url)


```shell
abradley@kolmogorov:~$ nvcc hello.cu -o hello.out
abradley@kolmogorov:~$ ./hello.out
Hello World!
```

**Comment by pavanky** [comment]
@bradleyashton I mean there should be a `helloworld_cuda` in `/home/abradley/arrayfire/build/examples`

Can you run that file and see what the output is?

**Comment by bradleyashton** [comment]
oh, sorry, here it is
```shell
abradley@kolmogorov:~/arrayfire/build/examples$ helloworld/helloworld_cuda
ArrayFire v3.5.0 (CUDA, 64-bit Linux, build e59cdbc)
Platform: CUDA Toolkit 7, Driver: 367.57
[0] Tesla K40c, 11440 MB, CUDA Compute 3.5
-1- Tesla K40c, 11440 MB, CUDA Compute 3.5
Create a 5-by-3 matrix of random floats on the GPU
A
[5 3 1 1]
    0.3990     0.5160     0.8831 
    0.6720     0.3932     0.0621 
    0.5339     0.2706     0.7089 
    0.1386     0.9455     0.9434 
    0.7353     0.1587     0.1227 

Element-wise arithmetic
B
[5 3 1 1]
    1.8885     1.9934     2.2727 
    2.1226     1.8831     1.5621 
    2.0089     1.7674     2.1510 
    1.6381     2.3108     2.3096 
    2.1708     1.6580     1.6224 

Negate the first three elements of second column
B
[5 3 1 1]
    1.8885    -1.9934     2.2727 
    2.1226    -1.8831     1.5621 
    2.0089    -1.7674     2.1510 
    1.6381     2.3108     2.3096 
    2.1708     1.6580     1.6224 

Fourier transform the result
C
[5 3 1 1]
         (9.8289,0.0000)          (-1.6751,0.0000)          (9.9178,0.0000) 
         (0.2648,-0.1720)          (-2.5026,5.7649)          (-0.3519,0.1506) 
         (-0.4579,0.3809)          (-1.6434,-1.7971)          (1.0748,-0.1153) 
         (-0.4579,-0.3809)          (-1.6434,1.7971)          (1.0748,0.1153) 
         (0.2648,0.1720)          (-2.5026,-5.7649)          (-0.3519,-0.1506) 

Grab last row
c
[1 3 1 1]
         (0.2648,0.1720)          (-2.5026,-5.7649)          (-0.3519,-0.1506) 
Scan Test
r
[16 4 1 1]
    2.0000     2.0000     2.0000     2.0000 
    2.0000     2.0000     2.0000     2.0000 
    2.0000     2.0000     2.0000     2.0000 
    2.0000     2.0000     2.0000     2.0000 
    2.0000     2.0000     2.0000     2.0000 
    2.0000     2.0000     2.0000     2.0000 
    2.0000     2.0000     2.0000     2.0000 
    2.0000     2.0000     2.0000     2.0000 
    2.0000     2.0000     2.0000     2.0000 
    2.0000     2.0000     2.0000     2.0000 
    2.0000     2.0000     2.0000     2.0000 
    2.0000     2.0000     2.0000     2.0000 
    2.0000     2.0000     2.0000     2.0000 
    2.0000     2.0000     2.0000     2.0000 
    2.0000     2.0000     2.0000     2.0000 
    2.0000     2.0000     2.0000     2.0000 

Scan
S
[16 4 1 1]
    2.0000     2.0000     2.0000     2.0000 
    4.0000     4.0000     4.0000     4.0000 
    8.0000     8.0000     8.0000     8.0000 
   16.0000    16.0000    16.0000    16.0000 
   32.0000    32.0000    32.0000    32.0000 
   64.0000    64.0000    64.0000    64.0000 
  128.0000   128.0000   128.0000   128.0000 
  256.0000   256.0000   256.0000   256.0000 
  512.0000   512.0000   512.0000   512.0000 
 1024.0000  1024.0000  1024.0000  1024.0000 
 2048.0000  2048.0000  2048.0000  2048.0000 
 4096.0000  4096.0000  4096.0000  4096.0000 
 8192.0000  8192.0000  8192.0000  8192.0000 
16384.0000 16384.0000 16384.0000 16384.0000 
32768.0000 32768.0000 32768.0000 32768.0000 
65536.0000 65536.0000 65536.0000 65536.0000 

Create 2-by-3 matrix from host data
D
[2 3 1 1]
    1.0000     3.0000     5.0000 
    2.0000     4.0000     6.0000 

Copy last column onto first
D
[2 3 1 1]
    5.0000     3.0000     5.0000 
    6.0000     4.0000     6.0000 

Sort A and print sorted array and corresponding indices
vals
[5 3 1 1]
    0.1386     0.1587     0.0621 
    0.3990     0.2706     0.1227 
    0.5339     0.3932     0.7089 
    0.6720     0.5160     0.8831 
    0.7353     0.9455     0.9434 

inds
[5 3 1 1]
         3          4          1 
         0          2          4 
         2          1          2 
         1          0          0 
         4          3          3 
```

**Comment by pavanky** [comment]
Ah crap, I forgot about another obvious reason for failure.

@bradleyashton @ranjanan is there a rogue `af_set_device` or the Julia equivalent being called in the code?

**Comment by bradleyashton** [comment]
just to reiterate the commands:
```julia
abradley@kolmogorov:~/arrayfire/build/examples$ AF_PRINT_ERRORS=1 julia
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.5.0 (2016-09-19 18:14 UTC)
 _/ |\__'_|_|_|\__'_|  |  
|__/                   |  x86_64-linux-gnu

julia> using ArrayFire

julia> getActiveBackend()
CUDA Backend

julia> function run(sz::Int)
           a = rand(AFArray{Float64}, sz, sz)
           b = rand(AFArray{Float64}, sz, sz)
           for i = 1:10
               r = a * b
               ArrayFire.eval(r)
           end
           sync()
       end
run (generic function with 1 method)

julia> run(100)
In function int cuda::DeviceManager::setActiveDevice(int, int)
In file src/backend/cuda/platform.cpp:428
CUDA Error (10): invalid device ordinal


ERROR: ArrayFire Error (998) : Internal error
 in throwAFerror(::Int32) at /home/abradley/.julia/v0.5/ArrayFire/src/error.jl:50
 in af_sync(::Int64) at /home/abradley/.julia/v0.5/ArrayFire/src/wrap.jl:1233
 in run(::Int64) at ./REPL[3]:8
```

**Comment by pavanky** [comment]
@bradleyashton can you add `AFInfo()` just after the for loop inside the `run` function ?

**Comment by bradleyashton** [comment]
```julia
julia> function run(sz::Int)
                  a = rand(AFArray{Float64}, sz, sz)
                  b = rand(AFArray{Float64}, sz, sz)
                  for i = 1:10
                      r = a * b
                      ArrayFire.eval(r)
                  end
       AFInfo()
       sync()
       end
WARNING: Method definition run(Int64) in module Main at REPL[8]:2 overwritten at REPL[10]:2.
run (generic function with 1 method)

julia> run(100)
ArrayFire v3.5.0 (CUDA, 64-bit Linux, build e59cdbc)
Platform: CUDA Toolkit 7, Driver: 367.57
[0] Tesla K40c, 11440 MB, CUDA Compute 3.5
-1- Tesla K40c, 11440 MB, CUDA Compute 3.5
In function int cuda::DeviceManager::setActiveDevice(int, int)
In file src/backend/cuda/platform.cpp:428
CUDA Error (10): invalid device ordinal


ERROR: ArrayFire Error (998) : Internal error
 in throwAFerror(::Int32) at /home/abradley/.julia/v0.5/ArrayFire/src/error.jl:50
 in af_sync(::Int64) at /home/abradley/.julia/v0.5/ArrayFire/src/wrap.jl:1233
 in run(::Int64) at ./REPL[10]:9
```

**Comment by pavanky** [comment]
Hmm this is completely weird. Does a simple `AFInfo()` followed by `sync()` fail ?

**Comment by bradleyashton** [comment]
```julia
julia> AFInfo()
ArrayFire v3.5.0 (CUDA, 64-bit Linux, build e59cdbc)
Platform: CUDA Toolkit 7, Driver: 367.57
[0] Tesla K40c, 11440 MB, CUDA Compute 3.5
-1- Tesla K40c, 11440 MB, CUDA Compute 3.5

julia> sync()
In function int cuda::DeviceManager::setActiveDevice(int, int)
In file src/backend/cuda/platform.cpp:428
CUDA Error (10): invalid device ordinal


ERROR: ArrayFire Error (998) : Internal error
 in throwAFerror(::Int32) at /home/abradley/.julia/v0.5/ArrayFire/src/error.jl:50
 in af_sync(::Int64) at /home/abradley/.julia/v0.5/ArrayFire/src/wrap.jl:1233
 in sync() at /home/abradley/.julia/v0.5/ArrayFire/src/backend.jl:89
```

**Comment by pavanky** [comment]
@bradleyashton if this happens in a fresh session of Jua, `sync` definitely the cause. 

@ranjanan Where is this function defined ?

**Comment by bradleyashton** [comment]
Ok, here is a fresh session (just to create MWE)
```julia
julia> quit()
abradley@kolmogorov:~/arrayfire/build/examples$ AF_PRINT_ERRORS=1 julia
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.5.0 (2016-09-19 18:14 UTC)
 _/ |\__'_|_|_|\__'_|  |  
|__/                   |  x86_64-linux-gnu

julia> using ArrayFire

julia> getActiveBackend()
CUDA Backend

julia> AFInfo()
ArrayFire v3.5.0 (CUDA, 64-bit Linux, build e59cdbc)
Platform: CUDA Toolkit 7, Driver: 367.57
[0] Tesla K40c, 11440 MB, CUDA Compute 3.5
-1- Tesla K40c, 11440 MB, CUDA Compute 3.5

julia> sync()
In function int cuda::DeviceManager::setActiveDevice(int, int)
In file src/backend/cuda/platform.cpp:428
CUDA Error (10): invalid device ordinal


ERROR: ArrayFire Error (998) : Internal error
 in throwAFerror(::Int32) at /home/abradley/.julia/v0.5/ArrayFire/src/error.jl:50
 in af_sync(::Int64) at /home/abradley/.julia/v0.5/ArrayFire/src/wrap.jl:1233
 in sync() at /home/abradley/.julia/v0.5/ArrayFire/src/backend.jl:89

julia> Pkg.status("ArrayFire")
 - ArrayFire                     0.0.3+             master
```

**Comment by bradleyashton** [comment]
looks like  its part of backend.jl
```julia
sync(b::Int64) at /home/abradley/.julia/v0.5/ArrayFire/src/backend.jl:93
  sync() at /home/abradley/.julia/v0.5/ArrayFire/src/backend.jl:88
```

and I managed to get this:

```shell
abradley@kolmogorov:~$ grep -A 10 sync .julia/v0.5//ArrayFire/src/backend.jl 
        getAvailableBackends, setBackend, getBackendId, sync,
        getActiveBackendId, getDeviceId, getVersionNumber

AF_BACKEND_DEFAULT = UInt32(0)
AF_BACKEND_CPU = UInt32(1)
AF_BACKEND_CUDA = UInt32(2)
AF_BACKEND_OPENCL = UInt32(4)

# Available only v3.3.0 onwards
function getActiveBackend()
    backend = getActiveBackendId()
--
function sync()
    b = getActiveBackendId()
    af_sync(b)
end

function sync(b::Int)
    af_sync(b)
end

# There is a af_get_device_id only in ArrayFire  > v3.3.x, possibly only in v3.4.0 onwards
```

**Comment by bradleyashton** [comment]
It seems to be defined in `backend.jl`
```shell
abradley@kolmogorov:~$ grep -A 10 sync .julia/v0.5//ArrayFire/src/backend.jl 
        getAvailableBackends, setBackend, getBackendId, sync,
        getActiveBackendId, getDeviceId, getVersionNumber

AF_BACKEND_DEFAULT = UInt32(0)
AF_BACKEND_CPU = UInt32(1)
AF_BACKEND_CUDA = UInt32(2)
AF_BACKEND_OPENCL = UInt32(4)

# Available only v3.3.0 onwards
function getActiveBackend()
    backend = getActiveBackendId()
--
function sync()
    b = getActiveBackendId()
    af_sync(b)
end

function sync(b::Int)
    af_sync(b)
end

# There is a af_get_device_id only in ArrayFire  > v3.3.x, possibly only in v3.4.0 onwards

function getDeviceId(a::AFArray)
    maj, min, pat = getVersionNumber()
    if maj < 3 || (maj == 3 && min < 4)
        throw("ArrayFire Version is v$maj.$min.$pat, which does not support this function. Install the latest version of ArrayFire.")
    end
    device = Base.Ref{Cint}(0)
```

am I right that `af_sync` is an `arrayfire` function? Now I am stumped...

If I run

```julia
julia> getActiveBackend()
Using libafcpu.so.3 from system path
Using libafopencl.so.3 from system path
Using libafcuda.so.3 from system path
CUDA Backend

julia> getActiveBackendId()
2
```

**Comment by ranjanan** [comment]
@pavanky is `af_set_device` called during `af_get_active_backend_id`?

**Comment by pavanky** [comment]
@ranjanan The ID that needs to be passed to sync is the deviceId! not the backendId!

**Comment by bradleyashton** [comment]
Thanks a lot for your help guys! It's not working yet for me, but sounds like you have a line on the problem...

**Comment by ranjanan** [comment]
@pavanky Thanks for pointing that out. I shall fix it. 

@bradleyashton In that case, does `AFInfo()` followed by `sync(0)` work for you?

**Comment by ViralBShah** [comment]
Let's make sure we do an ArrayFire min version check if we don't already do it.

**Comment by ranjanan** [comment]
@ViralBShah we already do it for that particular function. 

I think the best way to is to get the user to enter the device ID manually. The alternative would be creating a temporary and checking for its device ID.

**Comment by bradleyashton** [comment]
@ranjanan yes, 

```julia
julia> AFInfo()
ArrayFire v3.5.0 (CUDA, 64-bit Linux, build e59cdbc)
Platform: CUDA Toolkit 7, Driver: 367.57
[0] Tesla K40c, 11440 MB, CUDA Compute 3.5
-1- Tesla K40c, 11440 MB, CUDA Compute 3.5

julia> sync(0)
true
```

**Comment by ranjanan** [comment]
@bradleyashton Well, in that case, can you do a `Pkg.checkout("ArrayFire")` and then restart julia and run the benchmarks and see if they work for you?

**Comment by bradleyashton** [comment]
new error:
```julia
julia> include("benchmark.jl")
INFO: Warmup done!
INFO: Matmul
ERROR: LoadError: MethodError: no method matching sync()
Closest candidates are:
  sync(::Int64) at /home/abradley/.julia/v0.5/ArrayFire/src/backend.jl:88
 in matmul(::ArrayFire.AFArray{Float32,2}, ::ArrayFire.AFArray{Float32,2}) at /home/abradley/ArrayFire.jl/benchmarks/benchmark.jl:21
 in macro expansion at ./util.jl:226 [inlined]
 in benchmark() at /home/abradley/ArrayFire.jl/benchmarks/benchmark.jl:68
 in include_from_node1(::String) at ./loading.jl:488
while loading /home/abradley/ArrayFire.jl/benchmarks/benchmark.jl, in expression starting on line 107
```

**Comment by ranjanan** [comment]
@bradleyashton Are you sure you did a `Pkg.checkout("ArrayFire")` before  restarting Julia and checking? Because that has been changed to `sync(0)` in #132 and is now on the latest master.

**Comment by bradleyashton** [comment]
yes, just did it again to be sure

**Comment by ranjanan** [comment]
Can you paste line 21 of `benchmark.jl` here? Are you using the same file as is given in the repository?

**Comment by bradleyashton** [comment]
finally found it, I had another in home. Works!
```julia
julia> include("benchmark.jl")
WARNING: Method definition warmup() in module Main at /home/abradley/.julia/v0.5/ArrayFire/benchmarks/benchmark.jl:3 overwritten at /home/abradley/.julia/v0.5/ArrayFire/benchmarks/benchmark.jl:3.
WARNING: Method definition matmul(ArrayFire.AFArray, ArrayFire.AFArray) in module Main at /home/abradley/.julia/v0.5/ArrayFire/benchmarks/benchmark.jl:17 overwritten at /home/abradley/.julia/v0.5/ArrayFire/benchmarks/benchmark.jl:17.
WARNING: Method definition fast_fourier(ArrayFire.AFArray) in module Main at /home/abradley/.julia/v0.5/ArrayFire/benchmarks/benchmark.jl:25 overwritten at /home/abradley/.julia/v0.5/ArrayFire/benchmarks/benchmark.jl:25.
WARNING: Method definition cholesky(ArrayFire.AFArray) in module Main at /home/abradley/.julia/v0.5/ArrayFire/benchmarks/benchmark.jl:33 overwritten at /home/abradley/.julia/v0.5/ArrayFire/benchmarks/benchmark.jl:33.
WARNING: Method definition random() in module Main at /home/abradley/.julia/v0.5/ArrayFire/benchmarks/benchmark.jl:41 overwritten at /home/abradley/.julia/v0.5/ArrayFire/benchmarks/benchmark.jl:41.
WARNING: Method definition sorting(ArrayFire.AFArray) in module Main at /home/abradley/.julia/v0.5/ArrayFire/benchmarks/benchmark.jl:49 overwritten at /home/abradley/.julia/v0.5/ArrayFire/benchmarks/benchmark.jl:49.
WARNING: Method definition benchmark() in module Main at /home/abradley/.julia/v0.5/ArrayFire/benchmarks/benchmark.jl:58 overwritten at /home/abradley/.julia/v0.5/ArrayFire/benchmarks/benchmark.jl:58.
INFO: Warmup done!
INFO: Matmul
Time (CPU): 0.057478087
Time (GPU): 0.0062738954
INFO: FFT
Time (CPU): 0.062917006
Time (GPU): 0.0018234305
INFO: Cholesky
Time (CPU): 0.031043618
Time (GPU): 0.012306825
INFO: Rand
Time (CPU): 0.04071105
Time (GPU): 0.0009806825
INFO: Vec sort
Time (CPU): 0.054379303
Time (GPU): 0.0016460124
```

thanks a bunch!!

**Comment by ranjanan** [comment]
Alright then, this is fixed by #132
